src/utils/network.h

```diff
@@ -137,11 +137,31 @@ static inline int pg_utils_get_ipv6_l4(struct rte_mbuf *pkt, uint8_t *ip_type,
 			loop = false;
 		}
 	}
-	*ip_type = next_header;
-	*ip_payload = next_data;
+	if (!size) {
+		*ip_type = next_header;
+		*ip_payload = next_data;
+	} else {
+		*size += next_data - tmp;
```

Comment:
maybe a simpler solution would be to return the size instead of setting an input parameter.